Final touches for Theme Browser

kwonye opened the pull request:
* Fixed upgrade path (dropped table instead of deleting)
* Adding Drake to empty view and moving up to be visible while keyboard is up
* Active theme menu shows correct menu items

tonyr59h commented:
Using the search bar makes it really difficult to access the overflow menu on the list items. It disappears shortly after appearing if you have a search query.

* Search (i searched `dra`)
* Select menu on list item
* It disappears within 2s each time

HTC One, 5.1

tonyr59h commented:
When offline searching is still allowed but it won't filter the list. We have the theme names, can we perform an offline search of whatever data we have? If not I think it'd be best to disable searching when there is no connection.

tonyr59h commented:
Looking good! Just finished testing on the tablet (already done phone). I think it's set for a merge once those comments have been addressed.

tonyr59h commented on WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java:
```diff
@@ -470,10 +470,10 @@ protected void onPostExecute(final ArrayList<Theme> result) {
                     mFetchingThemes = false;
                     if (mThemeBrowserFragment != null && mThemeBrowserFragment.isVisible()) {
-                        mThemeBrowserFragment.mEmptyView.setText(R.string.no_themes);
+                        mThemeBrowserFragment.mEmptyTextView.setText(R.string.theme_no_search_result_found);
```
I think we should prefer a setter here rather than member variable access.

tonyr59h commented:
Can you clean up [here](https://github.com/wordpress-mobile/WordPress-Android/pull/3366/files#diff-15f797e2e7782e82ff2c89042bcb5eb5R92) by storing a reference to the `Resources` object?

tonyr59h commented:
[This check](https://github.com/wordpress-mobile/WordPress-Android/pull/3366/files#diff-1b8df52fda92e17c5761625d1c4fa553R103) isn't necessary anymore.

tonyr59h commented on WordPress/src/main/res/layout/theme_browser_fragment.xml:
```diff
@@ -32,17 +32,32 @@
+
+        <ImageView
+            android:id="@+id/drake_empty_results"
+            android:layout_width="400dp"
```
Use resource here.

tonyr59h commented:
I see `xhdpi` and `hdpi` Drake resources. Do we need `xxhdpi` and `xxxhdpi` resources as well?

kwonye commented:
> I see xhdpi and hdpi Drake resources. Do we need xxhdpi and xxxhdpi resources as well?

I've tested on the Nexus 9 (xxhdpi) and it looks great. These drawables are temporary until the svg-converter from the gradle update comes in (very soon)

> This check isn't necessary anymore.

This check is required, but I wrote it in a way that made the IDE not complain now. Good catch!

> When offline searching is still allowed but it won't filter the list. We have the theme names, can we perform an offline search of whatever data we have? If not I think it'd be best to disable searching when there is no connection.

Now when offline it'll do a local search. Thanks for this!

tonyr59h commented:
> This check is required

D'oh! Didn't notice the inheritance :+1:

tonyr59h commented:
> Now when offline it'll do a local search.

Woo hoo!

Found an issue with this, though :stuck_out_tongue: If you're online and have a search query when you go offline and then go back to not searching the current theme is lost (the current theme text says `Current Theme`). If you're offline before searching it doesn't seem to happen.

kwonye commented:
> Using the search bar makes it really difficult to access the overflow menu on the list items. It disappears shortly after appearing if you have a search query.

Addressed

> Found an issue with this, though :stuck_out_tongue: If you're online and have a search query when you go offline and then go back to not searching the current theme is lost (the current theme text says Current Theme). If you're offline before searching it doesn't seem to happen.

Addressed

tonyr59h commented:
Nice work! :shipit:
